modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala

```diff
@@ -178,12 +199,30 @@ abstract class SimpleProtocolBuilder[P](
     def flatMapErrors(
         fe: PartialFunction[Throwable, F[Throwable]]
     ): RouterBuilder[Alg, F] =
-      new RouterBuilder(service, impl, fe, middleware)
+      new RouterBuilder(
+        service,
+        impl,
+        fe,
+        middleware,
+        encodeErrorsBeforeMiddleware
+      )
 
     def middleware(
         mid: ServerEndpointMiddleware[F]
     ): RouterBuilder[Alg, F] =
-      new RouterBuilder[Alg, F](service, impl, errorTransformation, mid)
+      middleware(mid, encodeErrorsBeforeMiddleware = false)
+
+    def middleware(
+        mid: ServerEndpointMiddleware[F],
+        encodeErrorsBeforeMiddleware: Boolean
```

Comment:
let's make a dedicate method to set this up as opposed to an additional parameter on the middleware method.